### PR TITLE
[Bug 1259342] Fix datetimeformat jinja helpers

### PR DESCRIPTION
**Pull request by safwanrahman**
The issue should fix the jinja helper. Issue raised in #2819 

@Osmose @mythmon r?

### CODE REVIEW

File: kitsune/sumo/templatetags/jinja_helpers.py [hunk 1]
```diff
@@ -211,6 +211,14 @@ def datetimeformat(context, value, format='shortdatetime'):
     Returns a formatted date/time using Babel's locale settings. Uses the
     timezone from settings.py, if the user has not been authenticated.
     """
+
+    def get_user_timezone(user):
```
Comment (mythmon) [inline kitsune/sumo/templatetags/jinja_helpers.py]: Why did you move this code into an inner function? It seems a bit oddly placed, and doesn't really help with the readability.

File: kitsune/sumo/templatetags/jinja_helpers.py [hunk 2]
```diff
@@ -226,16 +234,18 @@ def datetimeformat(context, value, format='shortdatetime'):
...
+        if 'timezone' not in request.session:
+            if request.user.is_authenticated():
+                user_timezone = get_user_timezone(request.user)
+                if user_timezone is not None:
+                    convert_tzinfo = user_timezone
+            request.session['timezone'] = convert_tzinfo
+        else:
+            convert_tzinfo = request.session['timezone']
+
+    except AttributeError:
```
Comment (mythmon) [inline kitsune/sumo/templatetags/jinja_helpers.py]: I worry that this try/except is too broad. I'd rather see something like `if hasattr(request, 'session')`, which would be very specific. Catching generic `AttributeError`s, in particular, can have unwanted side effects as it can hide programmer errors, like misspelling a variable name.

**Comment by mythmon** [comment]
This should fix the problem, thanks! I think we can make this a little more robust before we merge it, though.

File: kitsune/sumo/templatetags/jinja_helpers.py (hunk 1 above)
Comment (safwanrahman) [inline kitsune/sumo/templatetags/jinja_helpers.py]: If not split into another inner function, the code looks messy and the code become hard to read. I think, getting the User timezone should be in another function.

**Comment by safwanrahman** [comment]
@mythmon updated. r?

**Comment by MikkCZ** [comment]
According to http://stackoverflow.com/questions/610883/how-to-know-if-an-object-has-an-attribute-in-python, try/catch will be faster, but this looks less as an _exception driven programming_.
